control-plane/connect-inject/handler.go

```diff
@@ -225,21 +231,92 @@ func (h *Handler) Handle(ctx context.Context, req admission.Request) admission.R
 		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error getting namespace metadata for container: %s", err))
 	}
 
-	// Add the init container that listens for the service and proxy service and sets up the Envoy configuration.
-	initContainer, err := h.containerInit(*ns, pod)
-	if err != nil {
-		h.Log.Error(err, "error configuring injection init container", "request name", req.Name)
-		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring injection init container: %s", err))
-	}
-	pod.Spec.InitContainers = append(pod.Spec.InitContainers, initContainer)
+	// Get service names from the annotation. If theres 0-1 service names, it's a single port pod, otherwise it's multi
+	// port.
+	annotatedSvcNames := h.annotatedServiceNames(pod)
+	multiPort := len(annotatedSvcNames) > 1
 
-	// Add the Envoy sidecar.
-	envoySidecar, err := h.envoySidecar(*ns, pod)
-	if err != nil {
-		h.Log.Error(err, "error configuring injection sidecar container", "request name", req.Name)
-		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring injection sidecar container: %s", err))
+	// For single port pods, add the single init container and envoy sidecar.
+	if len(annotatedSvcNames) == 0 || len(annotatedSvcNames) == 1 {
```

Comment:
I think that's equivalent?
```suggestion
	if len(annotatedSvcNames) < 2 {
```

Reply:
or maybe this could be `if !multiPort` and then the following `if multiPort` could be an `else`?

Reply:
made this an if !multiport + else!